.github/workflows/test.yml

```diff
@@ -10,41 +10,118 @@ on:
+
+      - name: Upload backtrace-library AAR
+        uses: actions/upload-artifact@v3
+        with:
+          name: backtrace-library-debug.aar
+          path: backtrace-library/build/outputs/aar/backtrace-library-debug.aar
+
+      - name: Upload backtrace-library Test APK
+        uses: actions/upload-artifact@v3
+        with:
+          name: backtrace-library-debug-androidTest.apk
+          path: backtrace-library/build/outputs/apk/androidTest/debug/backtrace-library-debug-androidTest.apk
+
+  test:
+    runs-on: ubuntu-latest
+    needs: build
+
+    strategy:
+      fail-fast: false
+      matrix:
+        test-apk: [example-app-debug-androidTest.apk, backtrace-library-debug-androidTest.apk]
+
+    steps:
+      - name: Download APK
+        uses: actions/download-artifact@v3
+        with:
+          name: example-app-debug.apk
+
+      - name: Download Test APK
+        uses: actions/download-artifact@v3
+        with:
+          name: ${{ matrix.test-apk }}
+
+      - name: Get Saucectl
+        run: curl -L https://saucelabs.github.io/saucectl/install | bash
+
+      - name: Install saucectl
+        uses: saucelabs/saucectl-run-action@v1
+        with:
+          skip-run: true
+
+      - name: Run Saucectl
+        run: |
+          saucectl configure --username ${{ secrets.SAUCE_USERNAME }} --accessKey ${{ secrets.SAUCE_ACCESS_KEY }}
+          # newest Android version
+          saucectl run espresso -c "" --name "From Github Actions" --app example-app-debug.apk --testApp ${{ matrix.test-apk }}  --emulator name="Android GoogleApi Emulator,platformVersion=12.0"
+          # oldest Android version
+          saucectl run espresso -c "" --name "From Github Actions" --app example-app-debug.apk --testApp ${{ matrix.test-apk }}  --emulator name="Android GoogleApi Emulator,platformVersion=5.1"
+          # real device
+          saucectl run espresso -c "" --name "From Github Actions" --app example-app-debug.apk --testApp ${{ matrix.test-apk }}  --device name="Samsung_Galaxy_S20_FE_5G_backtrace_us"
+          # --device name=".*,platformVersion=6.0.1"
+          # --device name="Samsung_Galaxy_S20_FE_5G_backtrace_us"
+          # --device name="Samsung_Galaxy_S22_Ultra_5G_backtrace_us"
+
+      - name: Run morgue
+        run: |
+          npm install backtrace-morgue --location=global
+          morgue --version || true
+          # Todo: currently just looks 15 minutes in the past and doesn't filter at all.
+          # --csv=crash.csv
+          morgue --token=${{ secrets.BACKTRACE_API_TOKEN }} --endpoint="https://yolo.sp.backtrace.io" list yolo/android-library --select timestamp --filter=timestamp,greater-than,`date +%s%3N -d "15 minutes ago"` || true
+
+#           if [ $(wc -l < crash.csv) -gt 0 ]; then
+#               echo "Success"
+#           else
+#               exit 1
+#           fi
+
+    #        runs-on: macos-latest # necessary for reactivecircus/android-emulator-runner@v2
+    #     - name: Emulator test
```

Comment:
do we need it?

Reply:
Maybe. Apilevel 16 is not supported on Sauce, which would be nice to test on. So we may have to boot an old school emulator.